From e78bc3de8ed1e7be4383b100e1839b317bd62e3e Mon Sep 17 00:00:00 2001
From: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
Date: Wed, 17 Aug 2016 15:17:37 +0700
Subject: [PATCH 4/6] gst145-v4l2src/v4l2bufferpool: Keep the information
 whether the videosink_buffer_creation_request query is supported in the
 downstream

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Signed-off-by: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
---
 sys/v4l2/gstv4l2bufferpool.c | 1 +
 sys/v4l2/gstv4l2src.c        | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index b3d7ca9..dc44cdb 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -1644,6 +1644,7 @@ gst_v4l2_buffer_pool_new (GstV4l2Object * obj, GstCaps * caps)
   pool->video_fd = fd;
   pool->obj = obj;
   pool->can_poll_device = TRUE;
+  pool->vsink_buf_req_supported = FALSE;
 
   pool->vallocator =
       gst_v4l2_allocator_new (GST_OBJECT (pool), obj->video_fd, &obj->format);
diff --git a/sys/v4l2/gstv4l2src.c b/sys/v4l2/gstv4l2src.c
index 93e8be3..f2a6db7 100644
--- a/sys/v4l2/gstv4l2src.c
+++ b/sys/v4l2/gstv4l2src.c
@@ -471,12 +471,17 @@ gst_v4l2src_decide_allocation (GstBaseSrc * bsrc, GstQuery * query)
   GstV4l2Src *src = GST_V4L2SRC (bsrc);
   gboolean ret = FALSE;
   GstBufferPool *pool;
+  GstV4l2Object *obj;
 
   if (pool) {
     GstStructure *config;
 
     config = gst_buffer_pool_get_config (pool);
 
+    gst_structure_get_boolean (config,
+        "videosink_buffer_creation_request_supported",
+        &GST_V4L2_BUFFER_POOL (obj->pool)->vsink_buf_req_supported);
+
     /*Check format of cap*/
     if(vin_format == GST_VIDEO_FORMAT_NV16) {
       /* support various metadata (neccesary when connect with videoconvert) */
-- 
1.9.1

